### ISSUE TITLE
apoc.custom.declareFunction fails when query calls other function.

### ISSUE THREAD

**Issue by solter** [neo4j-contrib/neo4j-apoc-procedures#2894]
## Expected Behavior (Mandatory)
After running
```
CALL apoc.custom.declareFunction(
    'f1(a1::INT) :: INT',
    'RETURN $a1 +1',
)
```
the following should succeed:
```
CALL apoc.custom.declareFunction(
    'f2(b1::INT) :: INT',
    'RETURN custom.g1($b1) +2'
);
```

## Actual Behavior (Mandatory)
The second statement fails with the message
```none
Failed to invoke procedure `apoc.custom.declareFunction`: Caused by: org.neo4j.exceptions.SyntaxException: Type mismatch for parameter 'b1': expected Integer but was String (line 1, column 26 (offset: 25))
"EXPLAIN RETURN custom.g1($b1) +2"
                          ^
```

Note: this bug is present in APOC 4.4.0.5, but not in APOC 4.4.0.3
Note: this behavior is also observed for the `apoc.custom.asFunction` method.

## How to Reproduce the Problem
Run the above statements.

### Versions

  - OS: Windows 10
  - Neo4j Desktop: version 1.4.15
  - Neo4j: 4.4.5
  - Neo4j-Apoc: 4.4.0.5

**Comment by solter** [comment]
I am uncertain if this is the result of the same underlying code as the bugs #2284, #2280 and #1871 that appear to be resolved by PR #2702

**Comment by vga91** [comment]
@solter 

There is a typo in your second query (`custom.g1`). But i guess you mean:
```
CALL apoc.custom.declareFunction(
    'f1(a1::INT) :: INT',
    'RETURN $a1 +1')
```
and 

```
CALL apoc.custom.declareFunction(
    'f2(b1::INT) :: INT',
    'RETURN custom.f1($b1) +2'
);
```

In this case, yes the error will be resolved by pr you indicated (`#2702`)

**Comment by solter** [comment]
@vga91 thanks for a) catching the typo and b) verifying that the pr fixes this bug
